src/main/java/io/openliberty/tools/common/plugins/util/BinaryScannerUtil.java

```diff
@@ -488,4 +522,38 @@ public String getMPLevel() {
             return mpLevel;
         }
     }
+
+    // A class to pass the invalid EE and MP parameters back to the caller.
+    public class IllegalTargetException extends Exception {
+        private static final long serialVersionUID = 1L;
+        String eeLevel;
+        String mpLevel;
+        IllegalTargetException(String eeLevel, String mpLevel) {
+            this.eeLevel = eeLevel;
+            this.mpLevel = mpLevel;
+        }
+        public String getEELevel() {
+            return eeLevel;
+        }
+        public String getMPLevel() {
+            return mpLevel;
+        }
+    }
+
+    // A class to pass the EE and MP parameters which do not work together back to the caller.
+    public class IllegalTargetComboException extends Exception {
```

Comment:
(Suggestion) This could extend the `IllegalTargetException` class above  since the code looks identical to get the `eeLevel` and `mpLevel`

Reply:
Good idea. I like to have separate exceptions to represent the two different concepts but no point duplicating all the code.